tests/test_io.cpp

```
@@ -0,0 +1,216 @@
+	auto read_back = read_raw<float>(tmp.path);
+	assert(read_back.size() == 3);
+	assert(read_back[0] == 1.0f);
+	assert(read_back[2] == 3.0f);
+
+	std::cout << "  raw_read_all: passed\n";
+}
+
+int main() {
+	std::cout << "Signal File I/O Tests\n";
+
+	test_wav_16bit_roundtrip();
```

Cppcheck (2.20.0)

[error] Unhandled exception thrown in function that is an entry point.

(throwInEntryPoint)